tests/src/whisk/core/cli/test/WskCoreBasicTests.scala

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.cli.test
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+import common.RunWskAdminCmd
+import common.TestHelpers
+import common.TestUtils
+import common.TestUtils.FORBIDDEN
+import common.TestUtils.NOT_FOUND
+import common.TestUtils.TIMEOUT
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol._
+import spray.json.pimpAny
+import whisk.core.entity.Subject
+import whisk.core.entity.WhiskPackage
+
+@RunWith(classOf[JUnitRunner])
+class WskCoreBasicTests
+    extends TestHelpers
+    with WskTestHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk(usePythonCLI = false)
+    val samplePackage = "samplePackage"
+    val sampleAction = samplePackage + "/sampleAction"
+    val sampleFeed = samplePackage + "/sampleFeed"
+    val wskadmin = new RunWskAdminCmd {}
+
```

Comment:
Sorry if I wasn't clear about where to put the "admin create" - by in the constructor, I meant here (line 47) rather than inside the test. Also there are a few ways of writing the tests a little bit more concisely and reusing existing code. I offer below two examples that rewrite the first two tests in the suite and leave the rest to you as an exercise.

```
    val otherNamespace = Subject().toString()
    val create = wskadmin.cli(Seq("user", "create", otherNamespace))
    val otherAuthkey = create.stdout.trim
    val otherWskProps = WskProps(namespace = otherNamespace, authKey = otherAuthkey)

    override def afterAll() = {
        withClue(s"failed to delete temporary namespace $otherNamespace") {
            wskadmin.cli(Seq("user", "delete", otherNamespace)).stdout should include("Subject deleted")
        }
    }

    behavior of "Wsk CLI"

    it should "reject deleting action in shared package not owned by authkey" in withAssetCleaner(otherWskProps) {
        (wp, assetHelper) =>
            assetHelper.withCleaner(wsk.pkg, samplePackage) {
                (pkg, name) => pkg.create(name, shared = Some(true))
            }

            assetHelper.withCleaner(wsk.action, sampleAction) {
                (action, name) =>
                    val file = Some(TestUtils.getTestActionFilename("empty.js"))
                    action.create(name, file, kind = Some("nodejs"), shared = Some(true))
            }

            val fullyQualifiedActionName = s"/$otherNamespace/$sampleAction"
            wsk.action.get(fullyQualifiedActionName)
            wsk.action.delete(fullyQualifiedActionName, expectedExitCode = FORBIDDEN)
    }

    it should "reject create action in shared package not owned by authkey" in withAssetCleaner(otherWskProps) {
        (wp, assetHelper) =>
            assetHelper.withCleaner(wsk.pkg, samplePackage) {
                (pkg, name) => pkg.create(name, shared = Some(true))
            }

            withAssetCleaner(wskprops) {
                (wp, assetHelper) =>
                    val fullyQualifiedActionName = s"/$otherNamespace/notallowed"
                    val file = Some(TestUtils.getTestActionFilename("empty.js"))
                    assetHelper.withCleaner(wsk.action, fullyQualifiedActionName, confirmDelete = false) {
                        (action, name) => action.create(name, file, expectedExitCode = FORBIDDEN)
                    }
            }
    }
```